metrics/collectors/economic.go

```diff
@@ -116,28 +131,46 @@ func (ec *EconomicCollector) Collect(ch chan<- prometheus.Metric) {
 		ch <- prometheus.MustNewConstMetric(ec.slashingStatus, prometheus.CounterValue, operatorIsFrozenFloat)
 	}
 
-	// collect delegatedShares metric
-	for _, strategyAddr := range ec.strategyAddrs {
-		sharesWei, err := ec.elReader.GetOperatorSharesInStrategy(context.Background(), ec.operatorAddr, strategyAddr)
-		if err != nil {
-			ec.logger.Error("Failed to get shares", "err", err)
-		} else {
-			// We'll emit all 3 units in case this is needed for whatever reason
-			// might want to change this behavior if this is emitting too many metrics
-			sharesWeiFloat, _ := sharesWei.Float64()
-			// TODO(samlaf): add the token name.. probably need to have a hardcoded dict per env (mainnet, goerli, etc)? Is it really that important..?
-			ch <- prometheus.MustNewConstMetric(ec.delegatedShares, prometheus.GaugeValue, sharesWeiFloat, strategyAddr.String(), "wei", "token")
-
-			sharesGweiFloat, _ := sharesWei.Div(sharesWei, big.NewInt(1e9)).Float64()
-			ch <- prometheus.MustNewConstMetric(ec.delegatedShares, prometheus.GaugeValue, sharesGweiFloat, strategyAddr.String(), "gwei", "token")
-
-			sharesEtherFloat, _ := sharesWei.Div(sharesWei, big.NewInt(1e18)).Float64()
-			ch <- prometheus.MustNewConstMetric(ec.delegatedShares, prometheus.GaugeValue, sharesEtherFloat, strategyAddr.String(), "ether", "token")
-		}
-	}
-
 	// collect registeredStake metric
 	// TODO(samlaf): implement this. probably have to call the BLSOperatorStateRetriever contract?
 	// probably should start using the avsregistry service instead of chainio clients so that we can
 	// swap out backend for a subgraph eventually
+	quorumNums, blsOperatorStateRetrieverOperator, err := ec.avsRegistryReader.GetOperatorsStakeInQuorumsOfOperatorAtCurrentBlock(context.Background(), ec.operatorId)
+	if err != nil {
+		ec.logger.Error("Failed to get operator stake", "err", err)
```

Comment:
somehow I like early return instead of else
```
if err != nil {
  // do something
  return;
}
//do another thing
```
wdyt?

Reply:
Agree but can't do that here because I want to emit the other metrics (so can't return early).

Reply:
are you talking about the commented code? I see.

Reply:
Ah I see your point. Indeed since its the last one I could do that. But I prefer to use the same structure across all the exported metrics. That way if I want to uncomment that code later, or add other metrics, I won't introduce a bug unknowingly.